src/main/java/org/broadinstitute/hellbender/engine/ReadShard.java

```diff
@@ -0,0 +1,182 @@
+package org.broadinstitute.hellbender.engine;
+
+import htsjdk.samtools.SAMSequenceDictionary;
+import htsjdk.samtools.util.Locatable;
+import org.broadinstitute.hellbender.engine.filters.ReadFilter;
+import org.broadinstitute.hellbender.utils.IntervalUtils;
+import org.broadinstitute.hellbender.utils.SimpleInterval;
+import org.broadinstitute.hellbender.utils.Utils;
+import org.broadinstitute.hellbender.utils.iterators.ReadFilteringIterator;
+import org.broadinstitute.hellbender.utils.read.GATKRead;
+
+import java.util.*;
+import java.util.stream.Collectors;
+import java.util.stream.StreamSupport;
+
+/**
+ * A class to represent a shard of reads data, optionally expanded by a configurable amount of padded data.
+ *
+ * The reads are lazily loaded by default (when accessing the reads via {@link #iterator}. Loading all the
+ * reads in the window at once is possible via {@link #loadAllReads}.
+ *
+ * The reads returned will overlap the expanded padded interval. It's possible to query whether they are within
+ * the main part of the shard via {@link #isContainedWithinShard} and {@link #startsWithinShard}.
+ *
+ * The reads in the window can be filtered via {@link #setReadFilter} (no filtering is performed by default).
+ */
+public final class ReadShard implements Iterable<GATKRead>, Locatable {
```

Comment:
Sorry, I didn't know about the removal. I think that `ReadShard` is a good name, the only problem is that I was confused with the other classes. In addition, if at the end the `ReadShard` is going to be hidden from the API-side, it won't be important.